compiler/src/main/kotlin/motif/compiler/XFunSpec.kt

```diff
@@ -50,23 +47,13 @@ object XFunSpec {
           env.requireType(method.returnType.typeName)
         }
 
-    val builder = overriding(methodElement)
-    builder.returns(javaToKotlinType(returnType))
-
-    var i = 0
-    val size = builder.parameters.size
-    val resolvedParameterTypes = method.parameterTypes
-    while (i < size) {
-      val parameter = builder.parameters[i]
-      val type = javaToKotlinType(resolvedParameterTypes[i])
```

Comment:
This is the real `type` we want, but we can only get it from `method` (which has the enclosing generic context), but not `methodElement` (which does not). Hence, we pass `method.parameterTypes` into the `overriding` function.